server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
Ideally, yes, but the hierarchy widget will only allow you to select items, not files.  When an item contains multiple files, you get a directory here.  This just my quick and dirty solution to avoid having to do a lot of work on the client side code.